Tests/SonarScanner.MSBuild.PreProcessor.Test/RoslynAnalyzerProviderTests.cs

```diff
@@ -230,147 +175,157 @@ private List<SonarRule> CreateRules() =>
+                Parameters = new Dictionary<string, string> { { "key", "value" } },
             },
-            new SonarRule("csharpsquid", "S1125", true),
-            new SonarRule("roslyn.wintellect", "Wintellect003", true),
-            new SonarRule("csharpsquid", "S1000", false)
+            new("csharpsquid", "S1125", true),
+            new("roslyn.wintellect", "Wintellect003", true),
+            new("csharpsquid", "S1000", false)
         ];
 
-    private string CreateTestFolders()
-    {
-        var rootFolder = TestUtils.CreateTestSpecificFolderWithSubPaths(TestContext);
-
-        // Create the binary and conf folders that are created by the bootstrapper
-        Directory.CreateDirectory(GetBinaryPath(rootFolder));
-        Directory.CreateDirectory(GetConfPath(rootFolder));
-
-        return rootFolder;
-    }
-
-    private static RoslynAnalyzerProvider CreateTestSubject(ILogger logger) =>
-        new(new MockAnalyzerInstaller(), logger);
-
     private static BuildSettings CreateSettings(string rootDir) =>
         BuildSettings.CreateNonTeamBuildSettingsForTesting(rootDir);
 
-    private static string GetConfPath(string rootDir) =>
-        Path.Combine(rootDir, "conf");
+    private class RoslynAnalyzerTestContext
+    {
+        public TestContext TestContext { get; set; }
+        public TestLogger Logger { get; } = new TestLogger();
+        public string RootDir { get; }
+        public IEnumerable<SonarRule> Rules { get; }
+        public string Language { get; }
+        public MockAnalyzerInstaller MockInstaller { get; }
+        public BuildSettings Settings { get; }
+        public ListPropertiesProvider SonarProperties { get; }
+        public RoslynAnalyzerProvider TestSubject { get; }
+        public AnalyzerSettings ActualSettings { get; set; }
+
+        public RoslynAnalyzerTestContext(TestContext testContext, string rootDir = null, IEnumerable<SonarRule> rules = null, List<string[]> analyzerPlugins = null, ListPropertiesProvider properties = null, string language = RoslynAnalyzerProvider.CSharpLanguage)
+        {
+            TestContext = testContext;
+            RootDir = rootDir;
+            Rules = rules;
+            Language = language;
+            Settings = rootDir is null ? null : CreateSettings(rootDir);
+            SonarProperties = properties ?? new ListPropertiesProvider();
+            MockInstaller = new MockAnalyzerInstaller(CreateAnalyzerPlugins(analyzerPlugins ?? new()));
+            TestSubject = new RoslynAnalyzerProvider(MockInstaller, Logger);
+        }
+
+        public RoslynAnalyzerTestContext(TestContext testContext, List<string[]> analyzerPlugins = null, ListPropertiesProvider properties = null)
+            : this(testContext, CreateTestFolders(testContext), CreateRules(), analyzerPlugins, properties) { }
 
-    private static string GetBinaryPath(string rootDir) =>
-        Path.Combine(rootDir, "bin");
+        public AnalyzerSettings ExecuteTest()
```

Comment:
It's weird to set and return it. Make this void and assert the analyzer settings from a property